tools/tesseratos/include/tesseratos/asset_explorer/popup.hpp

```diff
@@ -0,0 +1,48 @@
+/// @dir
+/// @brief @ref tesseratos-asset-explorer-popup utility file.
+
+/// @file
+/// @brief Utility function to show up a popup containing assets with given type.
+/// @ingroup tesseratos-asset-explorer-popup
+
+#pragma once
+
+#include <string>
+
+#include <cubos/core/ecs/system/accessors.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+
+#include <cubos/engine/assets/asset.hpp>
+#include <cubos/engine/assets/assets.hpp>
+
+namespace tesseratos
+{
+    /**
+     * @brief Displays a modal popup to select an asset of a specified type.
+     *
+     * @param title The title of the popup.
+     * @param selectedAssetBuffer Pointer to store the selected asset.
+     * @param type The type of asset to filter.
+     * @param assets The assets database to query.
+     * @return True if an asset is selected, false otherwise.
+     */
+    bool assetSelectionPopup(const std::string& title, cubos::engine::AnyAsset* selectedAssetBuffer,
```

Comment:
Rename `selectedAssetBuffer` to `selectedAsset` and make it a reference - no need to use pointers here.